src/core/server/legacy/legacy_service.ts

```diff
@@ -97,11 +96,11 @@ export class LegacyService implements CoreService {
     ).pipe(map(([http, csp]) => new HttpConfig(http, csp)));
   }
 
-  public async discoverPlugins(): Promise<LegacyServiceDiscoverPlugins> {
-    this.update$ = combineLatest(
+  public async setupLegacyConfig() {
+    this.update$ = combineLatest([
       this.coreContext.configService.getConfig$(),
-      this.coreContext.configService.atPath<PathConfigType>('path')
-    ).pipe(
+      this.coreContext.configService.atPath<PathConfigType>('path'),
+    ]).pipe(
```

Comment:
I had to keep a way to load the legacy configuration before calling the service's `setup` to perform config validation before setting up the services (see changes in `src/core/server/server.ts`)